src/script/filter.js

```diff
@@ -20,31 +23,51 @@ export class Filter {
     */
 
     filter () {
-        const renderer = new Renderer();
+        const renderer = new Renderer(this.data);
+        const filterByTag = new FilterByTag();
+        const filterByType = new FilterByType();
+        const filerByRating = new FilterByRating();
         const filteredList = { challenges: [],};
 
-        this.data.challenges.forEach(challenge => {
+        // Checks all filter elements if they are empty, if they all are empty (returning false) load all challenges
+        if ( 
+            filterByTag.checkDOM() ||
+            filterByType.checkDOM() ||
+            filerByRating.checkDOM()
+            ){
```

Comment:
This way of formatting braces in multiline if statements is not very common, and for that reason I would guess that it's difficult to keep it consistent within a team. In fact,  I see this line is not consistent even within this file (e.g. line 47, with the extra space) which strengthens my point.

Consistency is important. Maybe not as much at this very detailed level, but the best way to enforce consistency is to use auto formatting in your editor, and these little details tell me that you are not doing that.